feature/google/src/main/java/com/revenuecat/purchases/google/findBestOffer.kt

```diff
@@ -0,0 +1,115 @@
+@file:Suppress("MaximumLineLength", "MaxLineLength")
+package com.revenuecat.purchases.google
+
+import com.revenuecat.purchases.models.GoogleSubscriptionOption
+import com.revenuecat.purchases.models.PricingPhase
+import com.revenuecat.purchases.models.RecurrenceMode
+
+fun List<GoogleSubscriptionOption>.findBestOffer(): GoogleSubscriptionOption? {
+    val basePlan = this.firstOrNull { it.isBasePlan } ?: return null
+
+    val validOffers = this
+        .filter { !it.isBasePlan }
+        .filter { !it.tags.contains("rc-ignore-best-offer") }
+
+    return findLongestFreeTrial(validOffers) ?: findBestSavingsOffer(basePlan, validOffers) ?: basePlan
+}
+
+private fun findLongestFreeTrial(offers: List<GoogleSubscriptionOption>): GoogleSubscriptionOption? {
+    return offers.mapNotNull { offer ->
+        // Finds longest free pricing phase for an offer
+        offer.pricingPhases.filter { pricingPhase ->
+            pricingPhase.priceAmountMicros == 0L
+        }.map {
+            Pair(offer, parseBillPeriodToDays(it.billingPeriod)) // putting offer in pair here works but feels weird
+        }.maxByOrNull { it.second }
+    }.maxByOrNull { it.second }?.first
+}
+
+@Suppress("ReturnCount")
+private fun findBestSavingsOffer(
+    basePlan: GoogleSubscriptionOption,
+    offers: List<GoogleSubscriptionOption>
+): GoogleSubscriptionOption? {
+    val basePlanPricingPhase = basePlan.pricingPhases.firstOrNull() ?: return null
+
+    val longestOfferTotalDays = offers
+        .mapNotNull { subscriptionOption ->
+            subscriptionOption.pricingPhases.sumOf { it.totalNumberOfDays() ?: 0 }
+        }.maxByOrNull { it } ?: 0
+
+    if (longestOfferTotalDays < 0) {
+        return null
+    }
+
+    val periodDays = parseBillPeriodToDays(basePlanPricingPhase.billingPeriod)
+    val basePlanCostPerDay = basePlanPricingPhase.priceAmountMicros / periodDays
+
+    val offersWithCheapestPriceByDay = offers.map { offer ->
+
+        // Get normalized pricing for each phase
+        val pricePerDayPerPhase = offer.pricingPhases
+            .filter { it.recurrenceMode == RecurrenceMode.FINITE_RECURRING }
+            .mapNotNull { pricingPhase ->
+                val periodDays = parseBillPeriodToDays(pricingPhase.billingPeriod)
+
+                if (periodDays > 0) {
+                    val pricePerDay = pricingPhase.priceAmountMicros / periodDays
```

Comment:
Didn't we land on simply using the lowest $ value? https://revenuecat.slack.com/archives/C03M70X8DGF/p1674480990473239 
I understand that we don't need to compare periods, but it'd be just the one where the first non-free, non-baseplan price is lower

Reply:
Oh, we might have had a different understanding of "cheaper dollar value" then 😬 I thought that meant the "most savings" (cheapest) and not the "lowest" (cheapest)...

I have an issue with the "lowest" because there could be a "$1 for first day" for a $10/mo sub which is the lowest but not the best value if there is a "$5 for first month".

Thoughts?

Reply:
Updated!!